src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp

```diff
@@ -682,10 +683,17 @@ std::unique_ptr<WriteBufferFromFileBase> DiskObjectStorageTransaction::writeFile
 
     if (autocommit)
     {
-        create_metadata_callback = [tx = shared_from_this(), mode, path, blob_name] (size_t count)
+        create_metadata_callback = [tx = shared_from_this(), mode, path, blob_name](size_t count)
         {
             if (mode == WriteMode::Rewrite)
+            {
+                // Otherwise we will produce lost blobs which nobody points to
+                /// WriteOnce storages are not affected by the issue
+                if (!tx->object_storage.isWriteOnce() && tx->metadata_storage.exists(path))
```

Comment:
Do we need such changes in writeFileUsingBlobWritingFunction also?

Reply:
I think yes, let's do!

Reply:
Also `Unfreezer::removeFreezedPart` has similar branch `if (disk->supportZeroCopyReplication())`. Maybe it is worth to fix it too.